### ISSUE TITLE
Filter out known bad exposures

### ISSUE THREAD

**Issue by dkirkby** [dkirkby/legacyQA#8]
The prototype uses all 121,123 exposures listed in:
```
/global/project/projectdirs/cosmo/work/legacysurvey/dr8b/image-lists/image-list-decam-dr8.txt
```

This issue is to use a different list that has known bad exposures filtered out, so the expert labeling can focus on identifying bad exposures that are not already being automatically filtered by the existing quality cuts.

**Comment by dkirkby** [comment]
@moustakas recommends using:
```
/global/project/projectdirs/cosmo/work/legacysurvey/dr7/survey-ccds-dr7.fits.gz
```
which have all known quality cuts applied (and include the CP name <-> EXPNUM mapping?)

Known bad exposures are listed in https://github.com/legacysurvey/legacypipe/blob/master/py/legacyzpts/data/decam-bad_expid.txt

I see 3 classes of flagged bad exposure here, e.g.
```
#Flags: 1=possibly saturated; 2=clouds; 3=other source
583194  1 	between 10.7 and 12 twilight end of night
592619  2       lots of clouds, depth factor <<0.3
592791  3	insufficient exposure time, depth factor <0.3
```
There are also a lot of lines like:
```
346343 expfactor=0.208  seeing=1.400
```
@dstndstn Are these all considered bad, or are the cuts in expfactor and seeing still being optimized?

**Comment by moustakas** [comment]
Any exposure in this list is considered "bad" and should be rejected.  In the pipeline we read it with
https://github.com/legacysurvey/legacypipe/blob/master/py/legacyzpts/psfzpt_cuts.py#L274-L293
and then set the appropriate [`ccd_cuts` bit](https://github.com/legacysurvey/legacypipe/blob/master/py/legacyzpts/psfzpt_cuts.py#L17-L33) in the code [here](https://github.com/legacysurvey/legacypipe/blob/master/py/legacyzpts/psfzpt_cuts.py#L211).

**Comment by dkirkby** [comment]
Following up on @moustakas's suggestion in #6, is there a consensus that the best master list (available today) of CP outputs likely to be used in DR8 is:
```
/global/project/projectdirs/cosmo/work/legacysurvey/dr8/image-lists/dr8-ondisk-decam-v4.fits
```
I see files under six different paths there:
```
  725 CPDES82
42005 DECam_CP
  846 DECam_Reprocessed
31331 NonDECaLS
39335 NonDECaLS-DR8
30815 NonDECaLS-DR8-iY
```
Are these all going to be included in DR8?  There are 7K rows with duplicate EXPNUM (out of 145K): how should these be handled?

**Comment by moustakas** [comment]
Yes, that should be considered the definitive input image list, but you should cut on `qkeep=True` to resolve all duplicates.  Eddie's code has a bunch of logic to ascertain which exposure to process.

Going forward, the plan for DR9 is to clean up the files in the top-level directories--
https://github.com/legacysurvey/legacyzpts/issues/49

**Comment by dkirkby** [comment]
Thanks for the `qkeep` tip: I just confirmed that this removes all duplicates and selects 122K exposures.

**Comment by dkirkby** [comment]
After filtering on `qkeep == True`, there are still ~3K exposures that appear in https://github.com/legacysurvey/legacypipe/blob/master/py/legacyzpts/data/decam-bad_expid.txt. Is that expected?

**Comment by moustakas** [comment]
Yes, Eddie's scraper code (and the CP) knows nothing about the bad exposure list.  These images are/were taken into during observing, however, and will get a `ccd_cuts` bit and be rejected before Tractor gets run.

**Comment by dkirkby** [comment]
I am now using the list of exposures in:
```
/global/project/projectdirs/cosmo/work/legacysurvey/dr8/image-lists/dr8-ondisk-decam-v4.fits
```
filtered on `qkeep == True` and removing any exposures listed in https://github.com/legacysurvey/legacypipe/blob/master/py/legacyzpts/data/decam-bad_expid.txt.

The details are in `prepare/init.py` of this repo, which outputs:
```
Downloading decam-bad_expid.txt...
Read 2921 bad exposures from decam-bad_expid.txt.
Selected 121866 / 145057 exposures with qkeep==True.
Selected 119053 good exposures.
Created all.txt. To initialize jobs, use "cp all.txt todo.txt"
```

I believe this issue is ready to close now. Any last thoughts @moustakas?

**Comment by moustakas** [comment]
Unfortunately, @schlafly made another pass through the DECam exposures on-disk which resulted in more changes than any of us were expecting.  The new catalog is here:
```
/global/project/projectdirs/cosmo/work/legacysurvey/dr8/image-lists/dr8-ondisk-decam-v5.fits
```
I'm just a spectator on this part, but this *should* be it for DR8...

**Comment by dkirkby** [comment]
Ok, thanks for letting me know.  I didn't see any mention of this in [decam-chatter].  Are the v4 -> v5 changes documented somewhere?

I will work with v4 for now (since I already processed ~5% of the images), and then wait for the final DR8 list before updating.  Who is responsible for certifying the final DR8 list, btw? (pinging @sbailey since this came up in yesterday's telecon).

**Comment by moustakas** [comment]
* I've asked Eddie to send an update on-list. 

> Who is responsible for certifying the final DR8 list, btw?

Previously this was largely done by @arjundey and @djschlegel, with some iteration with the run manager / @mlandriau, but the various versions of the CP-processed data bit us into an inconsistency between old calibration files and more recently processed data.  

@schlafly has now taken the lead on this, so he and his `dr8-ondisk-[90prime,decam,mosaic]-v?.fits` lists are the final authority (as far as I'm concerned!) on the input image list.

**Comment by schlafly** [comment]
> Ok, thanks for letting me know. I didn't see any mention of this in [decam-chatter]. Are the v4 -> v5 changes documented somewhere?

Sorry, this is changing faster than I'd like.  We're now adding another night of data we had forgotten about, so there will be yet another version.  The temptation to add data since we're 'really at the end' is pretty heavy.

For your purposes, this really does not matter; the image lists are being tweaked at the <1% level for reasons that usually don't matter (biggest effect this time is different processing versions of the same files, which we are doing a better job excluding in -v5; 582 images).  The authoritative version of 75 images also changes between two different processings of those versions, now choosing the version with the later PLPROCID.

**Comment by dkirkby** [comment]
No problem.  I am using v4 for testing today and then will wait until a final image list has been blessed.
